src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -846,6 +845,20 @@ def _cleanUpExternalStore(url):
         blobService, containerName, _ = AzureJobStore._extractBlobInfoFromUrl(urlparse.urlparse(url))
         blobService.delete_container(containerName)
 
+@needs_azure
```

Comment:
The exponential back off already has some pretty comprehensive doctests. I don't think there's any functionality that is not accounted for by existing tests other than the added test here.